tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,72 @@
+import unittest
```

Comment:
Do we have test cases of retry?

Reply:
We haven't implemented the retry logic. It is handled in the SDK itself, and the `max_network_retries` was already present in the code. Hence we have not written testcases for retry.